server/api/hook.go

```diff
@@ -284,11 +284,26 @@ func branchFiltered(build *model.Build, remoteYamlConfigs []*remote.FileMeta) (b
 		}
 		log.Trace().Msgf("config '%s': %#v", remoteYamlConfig.Name, parsedPipelineConfig)
 
-		if parsedPipelineConfig.Branches.Match(build.Branch) {
-			return false, nil
+		// if was filtered by the constraints (event) continue
+		if !parsedPipelineConfig.Constraints.Event.Match(string(build.Event)) {
+			continue
+		}
+
+		// if was filtered by the constraints (branch) continue
+		if !parsedPipelineConfig.Constraints.Branch.Match(build.Branch) {
+			continue
+		}
+
+		// if was filtered by the branch (legacy) continue
+		if !parsedPipelineConfig.Branches.Match(build.Branch) {
+			continue
 		}
```

Comment:
Could we use something like the complete constraint checking as well? Or is this the problem that we haven't compiled the yml yet? We should probably find a cleaner way of doing this as otherwise a pipeline would be skipped by the code from `pipeline/frontend/yaml/compiler/compiler.go`, but would pass this code block.

```suggestion
		if !parsedPipelineConfig.Constraints.Match(c.Metadata) {
			continue
		}
```

Reply:
Should have a comment. This will also appear elsewhere but its nice.

Reply:
Wait. C. metadata is not valid here. So we cannot do that.